Feature/admin tutor allocation tutor details

stevenbister opened the pull request:
<!---
Thanks for creating a Pull Request 💖!
-->

### 🧹 Please check off the following before submitting:

- [x] Close issues
- [x] Request reviewers
- [x] Assign yourself
- [x] Add labels (e.g. `build:osc-ecommerce`)
- [x] Add project
- [x] Set project status
- [x] Add milestone

---

## ☑ List all github issues that this PR will close:

- Closes #1500 
- Closes #1501
- Closes #1502 

## 📝 Description

Apologies for the massive amount of files changed in this one 😅 Needed a bunch of new files and a few refactors to reflect updates to the api.

Adds the tutor profile drawer to the student allocation pages, this can be accessed by clicking on "view profile" on the tutor cards, when open this will update the url so it can be shared and accessed directly. I've set this up to load the profile data only when a drawer is open, this lets us load less data on initial load but will take a sec to load when the drawer opens, so there are a few skeleton loaders to show this.

There is a slight trade off here, that I can't quite work out how to get around -- if you open a tutor profile, then close the drawer it takes a second for the url params to sync. This means it is possible to navigate away before the search params sync, you have to be quick but it can happen. Not sure how to avoid this _and_ have the url state synced with the drawer so happy to hear any suggestions, but also don't think it's the end of the world.

## ⛳️ Current behaviour (updates)

N/a

## 🚀 New behaviour

- Renames `UserCard` to `TutorCard` to better reflect it's intent
- Queries data and pulls into `TutorCard`
- Adds some skeleton styles for drawer content
- Adds tutor profile drawer and inner components
- Updates the admin api to pull sen and course/student counts into the tutor profile
- Refactors `StudentSen` types to `UserSen` to reflect update to api
- 

## 💣 Is this a breaking change (Yes/No):

<!-- If Yes, please describe the impact and migration path for users. -->
> Please describe any breaking changes.


## 🧾 Final Checklist
<!-- Only mark what is applicable, Provide additional context if needed in the "Additonal Information". -->

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible cross-device impacts (iOS, Android)
- [x] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [x] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.


## 📝 Additional Information

> Any additional contextual information, such as relevant links/articles/documentation

<!--
 THANK YOU for your pull request!
-->

mattwendzina commented on packages/osc-academic-hub/app/components/Student/AssignedCourses.tsx:
```diff
@@ -0,0 +1,76 @@
+                            {course.course?.title}
+                        </span>
+                        <span className="o-grid__col o-grid__col--6 o-grid__col--3@desk-med t-font-s u-mb-0 u-text-bold">
+                            {course.tutor ? fullName(course.tutor as Tutor['profile']) : null}
```
Looks like you've got a ts error here

mattwendzina commented:
Looks like you've addressed most of the comments I'd left on the wrong PR 😂 

There was just the comment about some padding in mobile view and the console error that you didn't respond to:
```
TextInput contains an input of type search with both value and defaultValue props. Input elements must be either controlled or uncontrolled (specify either the value prop, or the defaultValue prop, but not both). Decide between using a controlled or uncontrolled input element and remove one of these props.
```

mattwendzina reviewed (commented):
Hey mate, sorry for leaving comments on the wrong PR! Looks like you've addressed most of the comments I'd made bar a couple which I've put on here.

stevenbister commented:
> Hey mate, sorry for leaving comments on the wrong PR! Looks like you've addressed most of the comments I'd made bar a couple which I've put on here.

Haha thanks mate, yeah I'll need to action those other comments :)
